includes/multi-currency/FrontendCurrencies.php

```diff
@@ -160,4 +190,34 @@ public function add_currency_to_cart_hash( $hash ): string {
 		$currency = $this->multi_currency->get_selected_currency();
 		return md5( $hash . $currency->get_code() . $currency->get_rate() );
 	}
+
+	/**
+	 * Checks whether currency code used for formatting should be overridden.
+	 *
+	 * @return bool
+	 */
+	public function should_override_currency_code(): bool {
+		return $this->utils->is_call_in_backtrace(
+			[
+				'WC_Shortcode_My_Account::view_order',
```

Comment:
Did not found a better check than these 2 ones. 
They should target specifically these two pages.